### PR TITLE
Migrate inferred schemas

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan @czue this enables inferred schemas for forms and uses them in the migrate command. can be read 🐟 

*Caveat in this PR*: This changes the InferredSchema to CaseInferredSchema and FormInferredSchema. If we merge this, we drop all InferredSchemas because I exclude them in the view. I don't think that's a huge deal since I don't think many people depend on InferredSchemas yet because we haven't migrated apps that needed to add things to the InferredSchema. However case imports and user case properties are added to the InferredSchema. The user case properties will be re-added as soon as a save on the user happens, so I'm not worried about that. We do have the potential to drop items that were imported in the case upload and not used in the application. If they are used in the export, they will remain in any export. I'm kind of OK dropping those, but that might be me being lazy and not thinking many people rely on them...

Last thing to do before this wretched migration is over is to revamp the feature flag that lets you add custom export properties to add them to the inferred schema.

cc: @emord 

Note: Reindex is on the meta database

### CODE REVIEW

File: corehq/couchapps/inferred_schemas_by_case_type_or_xmlns/views/view/map.js
Comment (dimagimon) [inline corehq/couchapps/inferred_schemas_by_case_type_or_xmlns/views/view/map.js]: Parsing error: Unexpected token (

File: corehq/apps/export/tests/test_export_conversion.py
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_conversion.py]: F821 undefined name 'ExportColumn'

File: corehq/apps/export/tests/test_export_conversion.py
Comment (dimagimon) [inline corehq/apps/export/tests/test_export_conversion.py]: E303 too many blank lines (3)

**Review by NoahCarnahan — APPROVED** [review]
Looks good to me.

With regards to losing case properties that only exist in case uploads: Do you have a ballpark number of how many domains this would effect? It sounds not great to drop this to me, but then again it won't matter if they do another upload or build out their apps to include those properties by the time they want to do an export.

Also just to make sure I'm understanding this right: Is the goal of adding `FormInferredSchema`s to support migrating remote apps?

**Comment by benrudolph** [comment]
> With regards to losing case properties that only exist in case uploads: Do you have a ballpark number of how many domains this would effect? It sounds not great to drop this to me, but then again it won't matter if they do another upload or build out their apps to include those properties by the time they want to do an export.

Fair will try and look into it and see if there's an easy way to tell

> Also just to make sure I'm understanding this right: Is the goal of adding FormInferredSchemas to support migrating remote apps?

That and support migrating other domains that are too old to have had build_ids attached to their form submissions

File: corehq/apps/export/utils.py
Comment (dimagimon) [inline corehq/apps/export/utils.py]: F401 'ROW_NUMBER_COLUMN' imported but unused

File: corehq/apps/export/utils.py
Comment (dimagimon) [inline corehq/apps/export/utils.py]: F401 'PathNode' imported but unused

**Comment by benrudolph** [comment]
✅ i think worth discussing in scale team meeting before merging this

**Comment by benrudolph** [comment]
after discussing with the scale team, we've decided to just merge this as is

**Comment by snopoke** [comment]
I assume you'll merge this when ready

**Comment by benrudolph** [comment]
👍 going to wait til formplayer settles down so don't have to wait on a preindex

**Comment by benrudolph** [comment]
merging and kicking off preindex

**Pull request merged**
